Add example.md (#108)
protofire/eth-cli: example.md

    - [`conf`](#conf)
      - [`conf:abi:add`](#confabiadd)
      - [`conf:abi:remove`](#confabiremove)
      - [`conf:address:add`](#confaddressadd)
      - [`conf:address:get`](#confaddressget)
      - [`conf:address:list`](#confaddresslist)
      - [`conf:address:remove`](#confaddressremove)
    - [`contract`](#contract)
      - [`contract:address`](#contractaddress)
      - [`contract:deploy`](#contractdeploy)
    - [`method`](#method)
      - [`method:decode`](#methoddecode)
      - [`method:encode`](#methodencode)
      - [`method:hash`](#methodhash)
      - [`method:send`](#methodsend)
      - [`method:send-transaction`](#methodsend-transaction)
    - [`networks`](#networks)
    - [`randomAddress`](#randomaddress)
    - [`transaction`](#transaction)
      - [`transaction:get`](#transactionget)
      - [`transaction:nop`](#transactionnop)
    - [`repl`](#repl)
  - [Back us](#back-us)

<!-- END doctoc generated TOC please keep comment here to allow auto update -->

## Examples

There are a lot of things that you can do with `eth-cli`, and we keep adding more. These are some of our favorites:

### Use help to learn how to play around with eth-cli

> ```

Reviewer: @janus How do you get this output?

Author: Please , which output are you referring?

If you want to see real  examples then try this.

```
eth abi:methods --help
Get the hash of each method in the given ABI.

USAGE
  $ eth abi:methods ABI

ARGUMENTS
  ABI  Contract ABI.

ALIASES
  $ eth me

EXAMPLE
  eth abi:methods ../contracts/proxy.abi

```

Note that this would not work `eth abi:methods -h`

### Start a REPL to interact with an RPC node

You can start a REPL to interact with the blockchain. By default it connects to `http://localhost:8545`, but you can use it with any chain.

```
$ eth repl --mainnet

> web3.eth.getBlockNumber()
6023243
```

This is a regular node REPL, but it includes a `web3` object that connects to the specified chain.

To find the host it is connected to

```
mainnet> let provider = web3.eth.currentProvider
mainnet> provider.host
'https://mainnet.infura.io'

```
```
mainnet> web3.eth.getProtocolVersion().then(console.log);
'0x3f'

```

```
web3.eth.getGasPrice().then(console.log);
'20000000000'

```

```
mainnet> web3.eth.getCode("0xD29F0b5b3F50b07Fe9a9511F7d86F4f4bAc3f8c4").then(console.log);
'0x6080604052600436106100af576000357c010000000000000000000000000000...
```

> You can also preload some contracts using a `<abi>@<address>` syntax:

Reviewer: ```suggestion
You can also preload contracts using a `<abi>@<address>` syntax:
```

```
$ eth repl --mainnet erc721@0x06012c8cf97bead5deae237070f9587f8e7a266d
> erc721.methods.name().call()
'CryptoKitties'
```

### Generate a random address

Do you need a placeholder for an address? Run `eth ra` (alias for `eth randomAddress`):

```
$ eth ra
{
  "address": "0x0e1c447DE1e870b5056e596C80e34da26569Deb2",
  "privateKey": "0x9093070def746fe7a3ad548b10dc0e6f861dc8753a41cd062de32c298517d4eb"
}
```

### Get info about the most popular chains

> Do you keep forgetting what is the id of rinkeby? Just run `eth networks

Reviewer: ```suggestion
Do you keep forgetting the id for the rinkeby network? Just run `eth networks
```

--table` to get information about each chain known by `eth-cli`.

```
$ eth networks --table
Id     Name        Url
1      Mainnet     https://mainnet.infura.io
3      Ropsten     https://ropsten.infura.io
4      Rinkeby     https://rinkeby.infura.io
30     RSK         https://public-node.rsk.co
31     RSK testnet https://public-node.testnet.rsk.co
42     Kovan       https://kovan.infura.io
77     Sokol       https://sokol.poa.network
99     POA         https://core.poa.network
100    xDAI        https://dai.poa.network
```

If you have [`jq`](https://stedolan.github.io/jq/) installed, you can do something like:

```
$ eth networks | jq .rinkeby.id
4
```

You can start repl with any of the networks.

```
eth repl --xdax
```
You can use url.

```
eth repl --url=https://dai.poa.network

```

Or

```
eth repl ./abis/myContract.json@0xaD2FA57bd95A3dfF0e1728686997F6f2fE67F6f9
eth repl erc20@0x34ee482D419229dAad23f27C44B82075B9418D31 
```

## Installation

> Install it globally:

Reviewer: ```suggestion
Install eth-cli globally:
```

`npm install -g eth-cli`

> You can also try it out with `npx`:

Reviewer: ```suggestion
You can also try installing eth-cli with `npx`:
```

}
$ eth randomAddress 3
{
  "address": "0xBb0144ba4fe2F10ACa2b11c8882183Ef643feCa9",
  "privateKey": "0x8218cd827a556fd924783e0d7a5d24a219cbd6ad984f4fbd0462927273feb774"
}
{
  "address": "0x04a08b29D4527B22e07c1aB3F8F393650452Fa25",
  "privateKey": "0xb44dd496a80a19f18d86a68890e1a52d84729d6adcc03b8a6dc5423f744e7389"
}
{
  "address": "0x3FE05C595c28CA01dC4cC4368362eA982FC23524",
  "privateKey": "0x286f60a2fab2bcbd1977af4fb3f502b0c92b13ea32fab7cda21fefd9d721e66a"
}
```

You can also specify a string to generate an address that starts with that prefix:

```
$ eth randomAddress 1 aaa
{
  "address": "0xaaa943F3cB8B5af6dB93A6BB594C6B6D202fC7Ab",
  "privateKey": "0x3ef9597d313e1f3663f160e03f9ce7b1909d18d713cb5cbe18ddfd1f15241a42"
}
```
Note this takes some seconds to complete.

### `transaction`

Commands to interact with transactions.

#### `transaction:get`

Print the transaction object for the given transaction hash.

```
$ eth tx:get --kovan 0x7f3f2967b93a7eded18071a596b7f3987c9575b6bae0a19399725c19e793517b
{
  "blockHash": "0xc85b64cdb940da308058b1f0ae800954a09287e2b8d30e71f02e1a4cf5b1a03b",
  "blockNumber": 11365979,
  "chainId": "0x2a",
  ...
```

> The transaction object will also include a `receipt` key, that will have the

Reviewer: ```suggestion
The transaction object also includes a `receipt` key, that has the
```
